tests/components/ipp/test_config_flow.py

```diff
@@ -0,0 +1,306 @@
+"""Tests for the IPP config flow."""
+import aiohttp
+from pyipp import IPPConnectionUpgradeRequired
+
+from homeassistant import data_entry_flow
+from homeassistant.components.ipp import config_flow
+from homeassistant.components.ipp.const import CONF_BASE_PATH, CONF_UUID
+from homeassistant.config_entries import SOURCE_USER, SOURCE_ZEROCONF
+from homeassistant.const import CONF_HOST, CONF_NAME, CONF_SSL
+from homeassistant.core import HomeAssistant
+
+from . import (
+    MOCK_USER_INPUT,
+    MOCK_ZEROCONF_IPP_SERVICE_INFO,
+    MOCK_ZEROCONF_IPPS_SERVICE_INFO,
+    init_integration,
+    load_fixture_binary,
+)
+
+from tests.test_util.aiohttp import AiohttpClientMocker
+
+
+async def test_show_user_form(hass: HomeAssistant) -> None:
+    """Test that the user set up form is served."""
+    result = await hass.config_entries.flow.async_init(
+        config_flow.DOMAIN, context={"source": SOURCE_USER},
+    )
+
+    assert result["step_id"] == "user"
+    assert result["type"] == data_entry_flow.RESULT_TYPE_FORM
+
+
+async def test_show_zeroconf_confirm_form(hass: HomeAssistant) -> None:
+    """Test that the zeroconf confirmation form is served."""
+    flow = config_flow.IPPFlowHandler()
+    flow.hass = hass
+    flow.context = {"source": SOURCE_ZEROCONF}
+    flow.discovery_info = {CONF_NAME: "EPSON123456"}
+
+    result = await flow.async_step_zeroconf_confirm()
```

Comment:
Always use the core interfaces to start the tests. Use `hass.config_entries.flow.async_init`.

Reply:
https://github.com/home-assistant/core/pull/33477